lib/logstash/outputs/webhdfs.rb

```diff
@@ -0,0 +1,265 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/outputs/base"
+require "stud/buffer"
+
+# Summary: Plugin to send logstash events to to files in HDFS via webhdfs 
+# restapi.  
+#
+# This plugin only has a mandatory dependency on the webhdfs gem from 
+# Kazuki Ohta and TAGOMORI Satoshi (@see: https://github.com/kzk/webhdfs).
+# Optional dependencies are zlib and snappy gem. 
+# No jars from hadoop are needed, thus reducing configuration and compatibility
+# problems.
+#
+# If you get an error like:
+#
+#     Max write retries reached. Exception: initialize: name or service not known {:level=>:error}
+#
+# make sure, that the hostname of your namenode is resolvable on the host running logstash. When creating/appending
+# to a file, webhdfs somtime sends a 307 TEMPORARY_REDIRECT with the HOSTNAME of the machine its running on.
+#
+# USAGE:
+# This is an example of logstash config:
+#
+# [source,ruby]
+# ----------------------------------
+# input {
+#   ...
+# }
+# filter {
+#   ...
+# }
+# output {
+#   webhdfs {
+#     server => "127.0.0.1:50070"         # (required)
+#     path => "/user/logstash/dt=%{+YYYY-MM-dd}/logstash-%{+HH}.log"  # (required)
+#     user => "hue"                       # (required)
+#   }
+# }
+# ----------------------------------
+#
+# Author: Bjoern Puttmann <b.puttmann@dbap.de> - dbap GmbH, Münster, Germany.
+
+class LogStash::Outputs::WebHdfs < LogStash::Outputs::Base
+  include Stud::Buffer
+
+  config_name "webhdfs"
+  milestone 1
+
+  MAGIC = "\x82SNAPPY\x0".force_encoding Encoding::ASCII_8BIT
+  DEFAULT_VERSION = 1
+  MINIMUM_COMPATIBLE_VERSION = 1
+
+  # The server name for webhdfs/httpfs connections.
+  config :host, :validate => :string, :required => true
+
+  # The server port for webhdfs/httpfs connections.
+  config :port, :validate => :number, :default => 50070
+
+  # The Username for webhdfs.
+  config :user, :validate => :string, :required => true
+
+  # The path to the file to write to. Event fields can be used here,
+  # as well as date fields in the joda time format, e.g.:
+  # ....
```

Comment:
Can you use in-place code highlighting for this? like: `/user/logstash/dt=%{+YYYY-MM-dd}/%{@source_host}-%{+HH}.log` for